src/main/java/com/hazelcast/aws/AwsDiscoveryStrategy.java

```diff
@@ -121,6 +125,14 @@ String getCurrentRegion(int connectionTimeoutSeconds, int connectionRetries) {
         return availabilityZone.substring(0, availabilityZone.length() - 1);
     }
 
+    private void validateRegion(String region) {
+        boolean isAwsRegion = AWS_REGION_PATTERN.matcher(region).matches();
+
+        if (!isAwsRegion) {
+            throw new AwsInvalidRegionException();
```

Comment:
1. I think that instead of creating your own exception, you can use the `InvalidConfigurationException` with the proper message.

2. I think it should include the configured `region` name.

Reply:
Fixed